src/Components/Server/src/ComponentHub.cs

```diff
@@ -45,23 +61,22 @@ internal sealed class ComponentHub : Hub
         /// </summary>
         public static PathString DefaultPath { get; } = "/_blazor";
 
-        /// <summary>
-        /// For unit testing only.
-        /// </summary>
-        // We store the circuit host in Context.Items which is tied to the lifetime of the underlying
-        // SignalR connection. There's no need to clean this up, it's a non-owning reference and it
-        // will go away when the connection does.
-        internal CircuitHost CircuitHost
+        // We store the CircuitHost through a *handle* here because Context.Items is tied to the lifetime
+        // of the connection. It's possible that a misbehaving client could cause disposal of a CircuitHost
+        // but keep a connection open indefinitely, preventing GC of the Circuit and related application state.
+        // Using a handle allows the CircuitHost to clear this reference in the background.
+        //
+        // See comment on error handling on the class definition.
```

Comment:
OK so if I follow correctly, outside a hub method call we're not allowed to mutate `Context.Items` directly, but we *can* mutate objects that its entries point to. So the handle technique is just a way of creating a mutable thing we can mutate at any time?

Reply:
Well I guess I'm wrong, since the handle never gets mutated.

Reply:
Correct. That is the point. 

It's there in a later commit. https://github.com/aspnet/AspNetCore/pull/12857/files#diff-8c77562947e4970a8d836d0df8be2b79R169